agent/src/softwarecontaineragent.h

```diff
@@ -296,6 +266,11 @@ class SoftwareContainerAgent
      * be passed to SoftwareContainer as a unique_ptr.
      */
     SoftwareContainerConfig m_containerConfig;
+
+    /*
+     * Responsible for parsing any dynamic values given when creating a container
+     */
+    ContainerConfigParser m_containerParser;
```

Comment:
I don't like this name very much. It can read a bit like "a parser for a container to use", or "parses a container". Maybe configParser or something like that?